Costco/__init__.py

```diff
@@ -0,0 +1,348 @@
+            pharmacy_id = pharmacy.telehippo_id
+            site_url = "https://apipharmacy.telehippo.com/api/c/{}/graphql".format(pharmacy_id)
+
+            query = ("""
+                query {
+                    cRetailerWithSetting (data:{slug:\"%s\"}) { 
+                        success,
+                        error,
+                        data { 
+                            retailer { 
+                                id,
+                                name,
+                                slug,
+                                startTime,
+                                endTime 
+                            } 
+                        }
+                    }
+                }
+            """ % pharmacy_id)
+
+            response = await session.post(site_url, json={"query": query})
+            try:
+                body = await response.json()
+                pharmacy.retailer_id = body['data']['cRetailerWithSetting']['data']['retailer']['id']
+            except (json.decoder.JSONDecodeError, KeyError, IndexError):
+                logging.error(
+                    "Failed to fetch data"
+                )
+                return []
+
+        notifications = []
+
+        for pharmacy in pharmacies:
+            retailer_id = pharmacy.retailer_id
+
+            appointments_url = "https://apipharmacy.telehippo.com/api/c/{}/graphql".format(retailer_id)
+            query = ("""
+                query { 
+                    onlineBookableAppointmentTypes (data:{retailerId:%s}) { 
+                        success,
+                        error,
+                        data {  
+                            services { 
+                                id,
+                                name,
+                                duration,
+                                bookingMode,
+                                description,
+                                intakeFormType,
+                                mainPageMode 
+                            } 
+                        } 
+                    }
+                }
+            """ % retailer_id)
+            
+            response = await session.post(appointments_url, json={"query": query})
+            try:
+                body = await response.json()
+                available_services = body['data']['onlineBookableAppointmentTypes']['data']['services']
+
+                for service in available_services:
+                    # This isn't super clear from the API, but I was able to derive the following:
+                    # bookingMode = 0 - Shows up on the site
+                    # bookingMode = 6 - Hidden from user-facing site
+                    # mainPageMode = 0 - Accepting Bookable Appointments
+                    # mainPageMode = 2 - Accepting Waitlist Signups
+                    if (service['bookingMode'] == 0 and service['mainPageMode'] == 0 and "covid" in service['name'].lower() and "vaccin" in service['name'].lower()):
```

Comment:
Was this meant to be "vaccine" not "vaccin"?

Reply:
Vaccin to catch "Vaccine" and "Vaccination"